File: JS/LeetC/g75/invertBinaryTree.py
import collections


class Solution:

    def invertTree(self, root):

        if not root:
            return root

        queue = collections.deque([root])

        while queue:
            current = queue.popleft()

            current.left, current.right = current.right, current.left

            if current.left:
                queue.append(current.left)

            if current.right:
                queue.append(current.right)

        return root

File: JS/LeetC/g75/invertBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */

function invertTree(root) {
  if (!root) {
    return root;
  }

  let right = invertTree(root.right);
  let left = invertTree(root.left);
  root.right = left;
  root.left = right;

  return root;
}


function invertTree(root) {
  if (!root) {
    return root;
  }

  const queue = [root];

  while (queue.length > 0) {
    current = queue.shift();
    [current.left, current.right] = [current.right, current.left];

    if (current.left) {
      queue.push(current.left);
    }

    if (current.right) {
      queue.push(current.right);
    }
  }

  return root;
}
